src/objective_types/twicedifferentiablehv.jl

```diff
@@ -0,0 +1,49 @@
+# Used for objectives and solvers where the gradient and Hessian is available/exists
+mutable struct TwiceDifferentiableHV{T,TDF,THv,TX} <: AbstractObjective
+    f
+    fdf
+    hv
+    F::T
+    DF::TDF
+    Hv::THv
+    x_f::TX
+    x_df::TX
+    x_hv::TX
+    v_hv::TX
+    f_calls::Vector{Int}
+    df_calls::Vector{Int}
+    hv_calls::Vector{Int}
+end
+iscomplex(obj::TwiceDifferentiableHV) = false
+
+# compatibility with old constructor
+function TwiceDifferentiableHV(f, fdf, h, x::TX, F::T, G::TG = similar(x), H::THv = similar(x)) where {T, TG, THv, TX}
+    x_f, x_df, x_hv, v_hv = x_of_nans(x), x_of_nans(x), x_of_nans(x), x_of_nans(x)
+    TwiceDifferentiableHV{T,TG, THv, TX}(f, fdf, h,
+                                        copy(F), similar(G), copy(H),
+                                        x_f, x_df, x_hv, v_hv,
+                                        [0,], [0,], [0,])
+end
+
+function TwiceDifferentiableHV(f, fdf, h, x::AbstractVector{T}) where T
```

Comment:
AbstractVector vs AbstractArray?

Reply:
You're right. I was lazy and just wanted to move it here. It's still unexported in Optim so we can do whatever we want :) But yes, that should be generalized (and tested on the generalization!)